torch/csrc/jit/tensorexpr/kernel.h

```diff
@@ -210,7 +210,7 @@ class TensorExprKernel {
   std::vector<Tensor*> tensorOutputs_;
   std::unordered_map<int64_t, Tensor*> tensors_;
   std::unordered_map<int64_t, VarHandle> scalars_;
-  std::unique_ptr<CodeGen> codegen_;
+  std::unordered_map<size_t, std::unique_ptr<CodeGen>> codegenCache_;
```

Comment:
For such a small map, it might be better to use a small vector. Since this is on critical path, the small perf squeeze might be helpful. But that could be left for a TODO for now.